packages/data/src/namespace-store/index.js

```diff
@@ -187,7 +187,10 @@ function mapSelectors( selectors, store, registry ) {
  * @return {Object}           Actions mapped to the redux store provided.
  */
 function mapActions( actions, store ) {
-	const createBoundAction = ( action ) => ( ...args ) => store.dispatch( action( ...args ) );
```

Comment:
For me this was an intended behavior though. It was explicitely meant to address the issue solved by the PR.

Reply:
> For me this was an intended behavior though. It was explicitely meant to address the issue solved by the PR.

What is the intended behavior? I need to check again, but as I understand the bug to be is that nothing will be returned by promise middleware in the current `master` branch. The change on this line seeks only to preserve that. Otherwise, what is the expected return value when dispatching an action?

Reply:
I think the expected return value here is a promise that is resolved once all the action flow (controls, sync actions...) is finished.

Reply:
I'm wondering if we're getting mixed up on the expectations of a dispatch which occurs via the result of the public interface `wp.data.dispatch`, and that of the internal `store.dispatch`, where the latter does in-fact (and still, after these changes) return a promise.

https://github.com/WordPress/gutenberg/blob/b3961d06a3862fb04967b776f2f246dc1f642f0a/packages/data/src/namespace-store/index.js#L260

I tested again on `master`, and neither a synchronous nor control action returns anything through `wp.data.dispatch`:

```js
wp.data.dispatch( 'core/editor' ).editPost( { title: 'hello' } );
// undefined
wp.data.dispatch( 'core/editor' ).savePost();
// undefined
```

However, it _was_ the case that generator actions would return promises as of v5.3.0 .

```js
wp.data.dispatch( 'core/editor' ).editPost( { title: 'hello' } );
// undefined
wp.data.dispatch( 'core/editor' ).savePost();
// Promise {<pending>}
```

This aligns with @nerrad 's earlier comment. I suspect it changed as a result of #14634.

So, the question(s) then are: Was it intentional to return a `Promise` for `wp.data.dispatch` and not just the internal `store.dispatch`? If so, shall we restore that behavior? And should it return a Promise for the synchronous `editPost` as above, or would it be fine enough since it would be normalized to a resolved promise in the context of [await](https://tc39.github.io/ecma262/#await) or a [`then` return value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#Return_value) (but not directly `.then`-able itself without manually normalizing via [`Promise.resolve`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve)).

Reply:
It has been beneficial in our usage that action generators result in a returned promise as there are many use-cases where the result of a save/update could be utilized immediately in further dispatched actions specific to an implementation (as opposed to general via the action).  So I'm not opposed to leaving it as.  I do think having "regular" actions not returning anything is fine.

Reply:
> I suspect it changed as a result of #14634.

`wp.data.dispatch( 'somestore').someActionGenerator()` has returned a pending promise for quite a while now.  I don't think #14634 introduced it.

Reply:
> `wp.data.dispatch( 'somestore').someActionGenerator()` has returned a pending promise for quite a while now. I don't think #14634 introduced it.

I'm suggesting it did the opposite: It removed it.

Reply:
YIKES! It did!  I haven't tested my work against the latest GB for a while.  So things are broke :(